### PR TITLE
Add data directory to Local Links Manager releases

### PR DESCRIPTION
**Pull request by sihugh**
- Create `data` directory on deploy in `shared` directory and then symlink into `public/data` inside the current release.
- This is to hold the exported CSV containing local authority services links which will be used by data.gov.uk.

[Trello card](https://trello.com/c/dFFQRlsp/448-generate-local-links-csv-3)

Mobbed with @brenetic @issyl0

### CODE REVIEW

**Comment by h-lame** [comment]
I was sure that Capistrano had functionality for doing this sort of thing directly.  It does in cap 3 (which we're not using): `set(:linked_dirs) = fetch(:linked_dirs) + %(public/data)`.  It also does in cap 2.x (what we are using): as above but the variable is `:shared_children` - the problem is that it seems it has inconsistent behaviour across different 2.x versions, and might not even do anything in some early 2.x veresions.  Shame.  Note that our python and node apps have a separate recipe that defines explicitly what to do with `shared_children` and make it safe for use when deploying python and node apps - we've not done anything for ruby apps though.

In summary; this explicit approach is probably safest until we upgrade capistrano.  Merging.
